src/gen/utils.py

```diff
@@ -69,7 +69,7 @@ async def call_service_llm(eval_model, prompt, retry_num=10, job_num=None):
             )
 
             assessment_json = _parse_first_json_snippet(assessment)
-            assessment_json = _required_keys_exist(assessment_json)
+            assessment_json = _required_keys_exist(assessment_json, keys_to_check)
```

Comment:
This could also be turned into a nice decorator. This way we can keep the validation logic out of the core LLM calling logic.